### DISCUSSION TITLE
How to configure html server in lspconfig

### DISCUSSION THREAD

**Discussion by gengdz** [LazyVim/LazyVim#1280] [Q&A] [answered]
Here's my configuration, but it's not working：
```lua
{
  "neovim/nvim-lspconfig",
  opts = {
    autoformat = false,
    servers = {
      html = {
        filetypes = { "html" },
      },
    },
    setup = {
      html = function(_, opts)
        local capabilities = vim.lsp.protocol.make_client_capabilities()
        capabilities.textDocument.completion.completionItem.snippetSupport = true
        require("lspconfig").html.setup({
          on_attach = function() end,
          capabilities = capabilities,
        })
      end,
    },
  },

```

It's too hard for me to make it work!

![image](https://github.com/LazyVim/LazyVim/assets/41912568/67443751-1c85-4596-8fb3-dfbf025f698d)
![image](https://github.com/LazyVim/LazyVim/assets/41912568/a29496e4-608d-4df0-b3e2-a4898c1ddfef)

**Comment by dpetka2001** [comment]
I only have 
```lua
{
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      html = {},
    },
  },
}
```
and it works out of the box. No need to setup `html` with `on_attach` and `capabilities`, since that is taken care of by `nvim-lspconfig` itself.

**Reply by gengdz** [reply]
I tried it. It didn't work

**Reply by dpetka2001** [reply]
Link to dotfiles?

**Reply by gengdz** [reply]
https://github.com/gengdz/dotfiles/blob/main/nvim/lua/plugins/lsp.lua

**Reply by dpetka2001** [reply, marked as answer]
You don't have completion inside the `script` tags correct? Where you want to write your JS commands. If that's the problem this happens because Neovim doesn't have good support with embedded languages (languages inside other languages). The solution for me was to add a plugin which opens a new buffer when you invoke it inside the `script` tags and then the `tsserver` gets correctly attached and offers you completion. You can take a look at my [dotfiles](https://github.com/dpetka2001/dotfiles/blob/3cd9413e1f32589b6bb7a090006f70677379d987/dot_config/nvim/lua/plugins/inline_edit.lua) for the plugin that does this. You must also have `tsserver` installed, because `html` doesn't do this.

**Reply by gengdz** [reply]
It worked. Thank you very much
